syntax = "proto3";

package hapi;

service GreeterService {
    rpc SayHello (HelloRequest) returns (HelloReply) {}

}

service ProfileService {

  rpc otpAuthenticate (OtpAuth) returns (OtpAuthResponse);
  rpc addsalesman (AddCustomerReq) returns (Response);

//   rpc DeleteUser(LogInRequest)  returns (Response);
}


message HelloRequest{
 string name = 1;
}

message HelloReply{
 string message = 1;
}
message OtpAuth {
    int32 otp           = 1;  // OTP
    string email        = 2;  // Email used for authentication of the user
  }
  
  message OtpAuthResponse {
    Response resp           = 1;   // Response
    string user_name        = 2;
    string user_phone       = 3;
    string user_email       = 4;
    // string access_token   = 5;
    // string refresh_token  = 6;
  }
  message Response {
    bool   status_code      = 1;  // Status Code
    string status_message   = 2;  // Status Message
  }
message AddCustomerReq{
  string user_name      =1;
  string user_phone     =2;
  string user_email     =3;
  string  company_id    =4;
  string role           =5;
}
